### DISCUSSION TITLE
Loading .duckdb file into memory

### DISCUSSION THREAD

**Discussion by makarr** [duckdb/duckdb#12506] [General]
Hi all, what's the simplest way to load a persistent db into memory? I'd like to use the in-memory version but keep a persistent copy as backup. The problem is, I don't see an easy way to import it in the documentation. .csv and .parquet are supported but apparently not the native format? What am I missing?

**Comment by Tishj** [comment]
Using ATTACH and COPY FROM DATABASE this can be done

Something like:
```sql
ATTACH 'path/to/persistent.db' other;
COPY FROM DATABASE other TO memory;
DETACH other;
```

Given the shell is started as `./duckdb` to create an in-memory session

**Comment by carlopi** [comment]
```
D ATTACH 'path/to/database.db' AS persistent_db (READ_ONLY);
D COPY FROM DATABASE persistent_db TO memory;
D DETACH persistent_db;
```

**Reply by carlopi** [reply]
What @Tishj says basically.

And you can do also the trip back (your call if overriding, or writing to other file).

**Comment by makarr** [comment]
Thanks for the quick answers, very impressed with the project so far
